packages/commonwealth/client/styles/components/component_kit/new_designs/CWTextInput.scss

```diff
@@ -0,0 +1,118 @@
+@import '../../../shared';
+
+@mixin iconPositionStyles {
+  position: absolute;
+  top: 50%;
+  transform: translateY(-50%);
+}
+
+.MessageRow {
+  display: flex;
+  align-items: center;
+
+  .icon {
+    display: grid;
+    align-items: center;
+    justify-content: center;
+
+    &.success > svg {
+      color: $green-600;
+    }
+    
+    &.failure > svg {
+      color: $rorange-500;
+    }
+  }
+
+  .feedback-message-text.Text {
+    display: grid;
+    align-items: center;
+    justify-content: center;
+    text-align: left;
+    margin-left: 8px;
+    
+    &.success {
+      color: $green-600;
+    }
+    
+    &.failure {
+      color: $rorange-500;
+    }
+  }
+}
+
+.TextInput {
+  display: flex;
+  flex-direction: column;
+
+  .input-and-icon-container {
+    display: flex;
+    align-items: center;
+    position: relative;
+    flex: 1;
+    --input-width: 240px;
+    width: var(--input-width);
+
+    &.fullWidth {
+      width: 100%;
+    }
+
+    input {
+      @include inputStyles;
+      width: var(--input-width);
```

Comment:
not sure why we introduce variables here? in this case `input` will always be `width: 100%`. `input-and-icon-container` will be 240 by default and might be overridden with `fullWidth` class.

Reply:
[Removed](https://github.com/hicommonwealth/commonwealth/commit/42cdff4965a362fc768f44280d56af5dc99f0027).